.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,57 @@
+name: Run Tests
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - qwerty/**
+      - qwerty_mlir/**
+      - test/**
+      - .github/**
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v4
+      - name: Install build dependencies
+        run: sudo apt install -y ninja-build python3-dev libffi-dev libzstd-dev libtinfo-dev
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install python dependencies
+        run: python -m pip install --upgrade pip setuptools
+      - name: Clone Tweedledum
+        uses: actions/checkout@v4
+        with:
+          repository: gt-tinker/tweedledum
+          ref: qwerty
+          path: tpls/tweedledum
+      - name: Clone QIR-Runner
+        uses: actions/checkout@v4
+        with:
+          repository: gt-tinker/qir-runner
+          ref: qwerty
+          path: tpls/qir-runner
+          ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
+      - name: Clone gtest
+        uses: actions/checkout@v4
+        with:
+          repository: google/googletest
+          ref: d83fee138a9ae6cb7c03688a2d08d4043a39815d
+          path: tpls/googletest
+      - name: Setting up LLVM and MLIR
+        run: |
+          wget https://github.com/gt-tinker/qwerty-llvm-builds/releases/download/v19.1.6/llvm_mlir_rel_v19_1_6_x86_linux.tar.xz
+          tar -xJvf llvm_mlir_rel_v19_1_6_x86_linux.tar.xz
+          echo "$GITHUB_WORKSPACE/llvm19/bin" >> $GITHUB_PATH
+      - name: Run the tests in virtual environment
+        run: |
+          export MLIR_DIR="$GITHUB_WORKSPACE/llvm19/lib/cmake/mlir"
+          export QWERTY_BUILD_TESTS=true
+          python3 -m venv venv
+          . venv/bin/activate
+          pip install -v .
+          pip install numpy
+          test/run-tests.sh
```

Comment:
This should be updated to follow the README. However, I don't know if `maturin develop` is the right move in a CI script... I know maturin has some kind of CI script generation you could look at